packages/slate-react/src/components/editor.tsx

```diff
@@ -0,0 +1,518 @@
+  renderElement?: (props: CustomElementProps) => JSX.Element
+  renderMark?: (props: CustomMarkProps) => JSX.Element
+  value: Value
+  [key: string]: any
+}) => {
+  const {
+    decorate = defaultDecorate,
+    editor,
+    value,
+    onChange,
+    placeholder,
+    readOnly = false,
+    renderAnnotation,
+    renderDecoration,
+    renderElement,
+    renderMark,
+    style = {},
+    ...attributes
+  } = props
+
+  if (!Value.isValue(value)) {
+    throw new Error(
+      `The \`value=\` prop you passed to <Editor> was not a valid Slate value: ${JSON.stringify(
+        value
+      )}`
+    )
+  }
+
+  const ref = useRef<HTMLDivElement>(null)
+
+  // Update internal state on each render.
+  editor.onChange = onChange
+  editor.value = value
+  PLACEHOLDER.set(editor, placeholder)
+  IS_READ_ONLY.set(editor, readOnly)
+
+  // Keep track of some state for the event handler logic.
+  const state = useMemo(
+    () => ({
+      isComposing: false,
+      isUpdatingSelection: false,
+      latestElement: null as DOMElement | null,
+    }),
+    []
+  )
+
+  // Update element-related weak maps with the DOM element ref.
+  useLayoutEffect(() => {
+    if (ref.current) {
+      EDITOR_TO_ELEMENT.set(editor, ref.current)
+      NODE_TO_ELEMENT.set(value, ref.current)
+      ELEMENT_TO_NODE.set(ref.current, value)
+    } else {
+      NODE_TO_ELEMENT.delete(value)
+    }
+  })
+
+  // Attach a native DOM event handler for `selectionchange`, because React's
+  // built-in `onSelect` handler doesn't fire for all selection changes. It's a
+  // leaky polyfill that only fires on keypresses or clicks. Instead, we want to
+  // fire for any change to the selection inside the editor. (2019/11/04)
+  // https://github.com/facebook/react/issues/5785
+  useLayoutEffect(() => {
+    window.document.addEventListener('selectionchange', onDOMSelectionChange)
+
+    return () => {
+      window.document.removeEventListener(
+        'selectionchange',
+        onDOMSelectionChange
+      )
+    }
+  }, [])
+
+  // Attach a native DOM event handler for `beforeinput` events, because React's
+  // built-in `onBeforeInput` is actually a leaky polyfill that doesn't expose
+  // real `beforeinput` events sadly... (2019/11/04)
+  // https://github.com/facebook/react/issues/11211
+  useLayoutEffect(() => {
+    if (ref.current) {
+      // @ts-ignore The `beforeinput` event isn't recognized.
+      ref.current.addEventListener('beforeinput', onDOMBeforeInput)
+    }
+
+    return () => {
+      if (ref.current) {
+        // @ts-ignore The `beforeinput` event isn't recognized.
+        ref.current.removeEventListener('beforeinput', onDOMBeforeInput)
+      }
+    }
+  }, [])
+
+  // Whenever the editor updates, make sure the DOM selection state is in sync.
+  useLayoutEffect(() => {
+    const { selection } = value
+    const domSelection = window.getSelection()
+
+    if (state.isComposing || !domSelection || !editor.isFocused()) {
```

Comment:
I believe you also need to check if `domSelect.type` is [`None`](https://developer.mozilla.org/en-US/docs/Web/API/Selection/type), otherwise, there is no range to select below.